Source/Core/VideoBackends/Software/TransformUnit.cpp

```diff
@@ -253,81 +248,58 @@ static void LightColor(const Vec3 &pos, const Vec3 &normal, u8 lightNum, const L
 			float cosAtt = light->cosatt.x + (light->cosatt.y * attn) + (light->cosatt.z * attn * attn);
 			float distAtt = light->distatt.x + (light->distatt.y * dist) + (light->distatt.z * dist2);
 			attn = SafeDivide(std::max(0.0f, cosAtt), distAtt);
+			break;
 		}
-		else if (chan.attnfunc == 1) // specular
-		{
-			// donko - what is going on here?  655.36 is a guess but seems about right.
-			attn = (light->pos * normal) > -655.36 ? std::max(0.0f, (light->dir * normal)) : 0;
-			ldir.set(1.0f, attn, attn * attn);
-
-			float cosAtt = std::max(0.0f, light->cosatt * ldir);
-			float distAtt = light->distatt * ldir;
-			attn = SafeDivide(std::max(0.0f, cosAtt), distAtt);
-		}
-		else
-		{
+		default:
 			PanicAlert("LightColor");
-			return;
-		}
-
-		switch (chan.diffusefunc)
-		{
-			case LIGHTDIF_NONE:
-				AddScaledIntegerColor(light->color, attn, lightCol);
-				break;
-			case LIGHTDIF_SIGN:
-				{
-					float difAttn = ldir * normal;
-					AddScaledIntegerColor(light->color, attn * difAttn, lightCol);
-				}
-				break;
-
-			case LIGHTDIF_CLAMP:
-				{
-					float difAttn = std::max(0.0f, ldir * normal);
-					AddScaledIntegerColor(light->color, attn * difAttn, lightCol);
-				}
-				break;
-			default: _assert_(0);
-		}
 	}
+
+	float difAttn = ldir * normal;
+	switch (chan.diffusefunc)
+	{
+		case LIGHTDIF_NONE:
+			AddScaledIntegerColor(light->color, attn, lightCol);
+			break;
+		case LIGHTDIF_SIGN:
+			AddScaledIntegerColor(light->color, attn * difAttn, lightCol);
+			break;
+		case LIGHTDIF_CLAMP:
+			difAttn = std::max(0.0f, difAttn);
+			AddScaledIntegerColor(light->color, attn * difAttn, lightCol);
+			break;
+		default: _assert_(0);
+	}	
 }
 
 static void LightAlpha(const Vec3 &pos, const Vec3 &normal, u8 lightNum, const LitChannel &chan, float &lightCol)
 {
 	const LightPointer *light = (const LightPointer*)&xfmem.lights[lightNum];
+	
+	Vec3 ldir = light->pos - pos;
+	float attn = 1.0f;
```

Comment:
It didn't appear to be needed in the games I was testing, but now that the code is refactored, both Light and LightAlpha will have the same result.  The other thing in question is if this is just relevant for LIGHTATTN_DIR or should all Attn modes have this fallback.  I think for now, unless there is a clear example, we should just leave it as this so nothing breaks.